src/Plugin/search_api/processor/IslandoraObjectImageDiscovery.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+namespace Drupal\dgi_image_discovery\Plugin\search_api\processor;
+
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\dgi_image_discovery\ImageDiscoveryInterface;
+use Drupal\node\NodeInterface;
+use Drupal\search_api\Datasource\DatasourceInterface;
+use Drupal\search_api\Item\ItemInterface;
+use Drupal\search_api\Processor\ProcessorPluginBase;
+use Drupal\search_api\Processor\ProcessorProperty;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Get the styled image url for the islandora node.
+ *
+ * @SearchApiProcessor(
+ *   id = "islandora_object_image_discovery",
+ *   label = @Translation("Islandora Object Image Discovery"),
+ *   description = @Translation("Get the styled image url for the islandora node."),
+ *   stages = {
+ *     "add_properties" = 0,
+ *   },
+ *   locked = true,
+ *   hidden = true,
+ * )
+ */
+class IslandoraObjectImageDiscovery extends ProcessorPluginBase implements ContainerFactoryPluginInterface {
```

Comment:
Not sure about the naming of this. Any particular reason not to use the established name? (i.e., `DgiImageDiscovery` or `DiscoverygardenImageDiscovery` or something of the like?)